features/cellular/framework/device/CellularDevice.cpp

```diff
@@ -118,25 +118,26 @@ nsapi_error_t CellularDevice::attach_to_network()
 nsapi_error_t CellularDevice::create_state_machine()
 {
     nsapi_error_t err = NSAPI_ERROR_OK;
+    _nw = open_network(_fh);
```

Comment:
Now when this is not done only once as in the old code, it will endup increasing the ref count for network object, but the ref count is not decreased on stop.